tests/test-location.py

```diff
@@ -156,11 +158,34 @@ def quit_on(self, method_name):
         GLib.timeout_add_seconds(20, self.fail, 'Test timed out after ' +
                                  'waiting 20 seconds for D-Bus method call.')
 
+    def quit_on_event(self, event_uuid):
+        """Quit the main loop when RecordSingularEvent is called for @event_uuid.
+        Timeout after waiting for 20 seconds. Use like this:
+            self.quit_on_event(uuid.UUID('my-uuid'))
+            self.mainloop.run()
+            # now event my-uuid has been received
+        """
+        self._quit_on_uuid = event_uuid
+        self.quit_on('RecordSingularEvent');
+
     def handle_dbus_event_received(self, name, *args):
         if name == self._quit_on_method:
+            if (self._quit_on_uuid):
```

Comment:
I'm not sure what this does, so I usually use `if self._quit_on_uuid is not None`.